cpp/FileTransferAgent.cpp

```diff
@@ -139,6 +145,7 @@ void Snowflake::Client::FileTransferAgent::initFileMetadata(std::string *command
   m_FileMetadataInitializer.setAutoCompress(response.autoCompress);
   m_FileMetadataInitializer.setSourceCompression(response.sourceCompression);
   m_FileMetadataInitializer.setEncryptionMaterials(&response.encryptionMaterials);
+  m_FileMetadataInitializer.setRandomDev(m_useDevUrand);
```

Comment:
setRandomDev is internal func in cpp/FileMetadataInitializer.hpp which sets the real random device to be used and the Encryptor uses the device value from this func.

The other setRandomDeviceAsUrand() is invoked by filetransferagent and is used from ODBC and applications directly.